### ISSUE TITLE
Should error in predict_in_sample when start < d

### ISSUE THREAD

**Issue by axkod** [alkaline-ml/pmdarima#286]
When creating an ARIMA model with a differencing of more than 0 it should be impossible to get the prediction of the model for the index 0.

However, when calling predict_in_sample with start=0 on an ARIMA model with a differencing of 1 or more it will return a value for the index 0. And when calling this function with return_conf_int=True the confidences intervals for this specific first value are not coherent (extremely big values).

As explained at the end of this page: https://www.statsmodels.org/dev/generated/statsmodels.tsa.arima_model.ARIMAResults.predict.html#statsmodels.tsa.arima_model.ARIMAResults.predict
It might be a good idea to throw an error when trying to call the function with a start "out of range".

**Comment by pypinkfloydfan** [comment]
This is really interesting.  In a similar vein, I noticed today that when testing my ARIMA model (which also has a differencing of 1) and plotting the forecasts vs actual values, that all the predicted values seemed to be occurring 1 step later than the actual values in the graph.

All I'm doing is splitting the dataset 80:20 for train:test, then forecasting 1-step and rebuilding the model each time.

It would appear that I need to forecast an extra step and remove the very first value but I have no idea why.  

It makes me wonder if my goodness of fit tests (RMSE, MAPE, SMAPE) are accurate or not?

Can anyone help explain this please?

Thanks very much

**Comment by tgsmith61591** [comment]
@axkod I think that's a fair point. Maybe not an error, but certainly a warning.

@pypinkfloydfan Can you provide some reproducible code that exhibits what you're seeing?

**Comment by pypinkfloydfan** [comment]
Hi Taylor, 

Apologies for the late response, I have been trying to find my error (if there is one) but can't seem to figure it out.  

This is the relevant code block that seems to be generating lagged predictions in an ARIMA model of the form SARIMAX(1, 1, 1)x(1, 0, 1, 63):

    """
    trainpct = 0.8
    split = int(len(model1data) * trainpct) 
    train, test = model1data[0:split], model1data[split:]

    def forecast_one_step(model, stepsize=1):
        '''
        This function forecasts 'stepsize' step forwards of the given time series 'model' and returns the forecast and confidence intervals as two lists.
    
        Parameters:
        model: an ARIMA model from pmdarima built using a training set of data
        stepsize: how many steps forwards to forecast
    
        Return:
        list of forecasts, list of confidence intervals
        '''


        fc, conf_int = model.predict(n_periods=stepsize, return_conf_int=True)
        return (fc.tolist()[0], conf_int.tolist()[0])

        def forecast_all_steps(model, y_test, stepsize=1):
        '''
        This function is used in conjunction with 'forecast_one_step' to rebuild the model after each 'stepsize' has been forecast.  At present only written for stepsizes of 1
        In that way, the model uses the forecast data to redefine itself, creating a more accurate representation of how it would be used in real life
    
        Parameters:
        model: an ARIMA model from pmdarima built using a training set of data
        stepsize: how many steps forwards to forecast
        y_test: the test set of data to forecast
    
        Return:
        A list of forecasts and confidence intervals
        '''

        forecasts = list()
        confidence_intervals = list()
        testlen = len(y_test)
        t0 = time()
        out = Output()
        display(out)
    
        i=1
        for new_ob in y_test:
            out.clear_output()
            t1 = time()
            with out:
                print('Calculating forecast {} of {}'.format(i, testlen))
                print('Total time so far {:.2f} mins.'.format((t1-t0)/60))
            fc, conf = forecast_one_step(model, stepsize)
            forecasts.append(fc)
            confidence_intervals.append(conf)

        # Update the existing model 
        model.update(new_ob)
        i+=1
        
        return (forecasts, confidence_intervals)

    fc, ci = forecast_all_steps(sarimamodel, test)

Thanks very much for any guidance you can offer
